### PR TITLE
fix(rich-text-editor): DP-179947 use URL as display text when inserting link without text

### PR DESCRIPTION
**Pull request by braddialpad**
## Description

JIRA Link: https://dialpad.atlassian.net/browse/DP-179947

When a user inserts a link in the rich text editor without providing display text, the link is now visible in the editor using the URL itself as the display text.

## Summary of changes requested

- Fix `setLink` in `DtRichTextEditor` to fall back to the URL when display text is empty
- Add tests covering both the display-text-provided and display-text-empty cases

## Technical Details

**Root cause:** `setLink(linkInput, linkText, ...)` called `insertContent(linkText)` followed by `setTextSelection({ from, to: from + linkText.length })`. When `linkText` was `""`, this inserted nothing and created a zero-width selection, so the subsequent `setLink()` mark command had no range to apply to — leaving the editor with no visible link node.

**Fix:** `packages/dialtone-vue/components/rich_text_editor/rich_text_editor.vue` — introduce `const displayText = linkText || linkInput` and use it for both `insertContent` and `setTextSelection`, so the URL is used as the visible text when no display text is provided.

**Tests:** Added two cases to `rich_text_editor.test.js` under a new `setLink method` describe block — one for when display text is provided, one for when it is empty.

Use the URL as visible link text when no display text is provided to prevent zero-width selections. setLink now derives displayText = linkText || linkInput and uses it for insertContent and selection. Added tests covering both non-empty and empty display-text link insertion.

### CODE REVIEW

File: packages/dialtone-vue/components/rich_text_editor/rich_text_editor.test.js [hunk 1]
```diff
@@ -1633,6 +1633,37 @@ describe('DtRichTextEditor tests', () => {
...
+      });
+
+      describe('When display text is empty', () => {
+        it('should use the URL as the display text', async () => {
+          wrapper.vm.setLink('https://example.com', '', linkOptions, supportedProtocols, defaultPrefix);
+          await wrapper.vm.$nextTick();
+          const output = wrapper.vm.getOutput();
+          expect(output).toContain('href="https://example.com"');
+          expect(output).toContain('https://example.com');
+        });
```
[the remark this answers is not in the document]
Comment (braddialpad) [reply packages/dialtone-vue/components/rich_text_editor/rich_text_editor.test.js]: good suggestion, fix made.

**Review by iropolo — APPROVED** [review]
